Explorer/Assets/Scripts/Utility/ThreadSafePool/ThreadSafeObjectPool.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using UnityEngine.Pool;
+
+namespace Utility.ThreadSafePool
+{
+    public class ThreadSafeObjectPool<T> : IObjectPool<T> where T: class
```

Comment:
I didn't realise IObjectPool wasn't thread-safe. To prevent thread contention through resource locks would it be better to use something like ThreadLocal ?

https://learn.microsoft.com/en-us/dotnet/api/system.threading.threadlocal-1?view=net-8.0

Reply:
I truly believe that the component pools should be shared between threads

Reply:
I agree. Given that the scenes are going to be run in different threads, the component pool is gonna be shared between threads.